packages/common/src/store/ui/stripe-modal/slice.ts

```diff
@@ -40,9 +42,11 @@ const slice = createSlice({
     cancelStripeOnramp: () => {},
     stripeSessionStatusChanged: (
       state,
-      action: PayloadAction<{ status: StripeSessionStatus }>
+      action: PayloadAction<{ session: StripeSessionData }>
     ) => {
-      state.stripeSessionStatus = action.payload.status
+      state.previousStripeSessionData = state.stripeSessionData
+      state.stripeSessionData = action.payload.session
+      state.stripeSessionStatus = action.payload.session.status
```

Comment:
Note on this: I opted _not_ to remove the separate status tracking because we reference it in a number of places and I wanted to _add_ analytics without potentially breaking other code. But willing to do the work of removing it if anyone feels strongly about it